pkgs/applications/virtualization/qemu/default.nix

```diff
@@ -86,10 +86,9 @@ stdenv.mkDerivation (finalAttrs: {
     # workaround, remove once this patch lands: https://lore.kernel.org/qemu-devel/20240805104921.4035256-1-hi@alyssa.is/
     ++ lib.optionals (hexagonSupport && stdenv.hostPlatform.isStatic) [ pcre2 ];
 
-  buildInputs = [ zlib glib pixman
-    vde2 lzo snappy libtasn1
-    gnutls nettle curl libslirp
-  ]
+  buildInputs = [ glib zlib ]
+    ++ lib.optionals (!minimal) [ dtc pixman vde2 lzo snappy libtasn1 gnutls nettle libslirp ]
+    ++ lib.optionals (!userOnly) [ curl ]
```

Comment:
I am not really sure how to verify this but disabling features we don't need is probably a good idea.

Reply:
What I did:
1. Verify manually that for `!minimal && !userOnly` (i.e. all the normal qemus like qemu, qemu-full, qemu_kvm…), only the order of inputs changes (which is hopefully irrelevant).
2. qemu-user working as intended is covered by a test added here
3. For qemu-utils:
   * `nix build github:NixOS/nixpkgs/{c46d43c625f161d3bd22052a5f8f9df703d81be8,f69411bcfb0dd3565a19667fdf509bbadb38f008}#qemu-utils`
   * The sizes of the output binaries are all the same: `ls -l result*/bin/* | cut -d\  -f5- | sort`, so it's unlikely that some larger functionality went missing
   * Looking e.g. through `nix run nixpkgs#diffoscope result*/bin/qemu-storage-daemon` shows some probably irrelevant differences in the length of some dummy strings (Probably due to `error: derivation '…-qemu-utils-9.0.2.drv' may not be deterministic: output '…-qemu-utils-9.0.2' differs` -.-)